tests/func/utils/support/mangakakalot.py

```diff
@@ -26,34 +26,39 @@ def home(self):
         self._driver.get('https://mangakakalot.com/')
 
     @retry(abort=True)
-    def load_random(self, predicate=None):
-        to_ignore = []
-
-        mangas = self._get_mangas()
-        while len(mangas) > 0:
-            manga = mangas.pop(random.randrange(len(mangas)))
-            chapters = manga.find_elements(by=By.CSS_SELECTOR, value='li>span>a')
-            if len(chapters) < 3:
-                continue
-            href = chapters[1].get_attribute('href')
-            if '://mangakakalot.com/' not in href or href in to_ignore:
-                continue
-            self._driver.get(href)
-            # Validate predicate if specified
-            if predicate and not predicate(self):
-                to_ignore.append(href)
-                mangas = self._get_mangas()
+    @check_predicate(RetriableError)
+    def load_random(self):
+        chapters_lists = [manga.find_elements(by=By.CSS_SELECTOR, value='li>span>a')
+                          for manga in self._get_mangas()]
+        def testable(cs):
```

Comment:
Please either put this inner function at the top of this function or surround it with empty lines.